src/slam_toolbox_common.cpp

```diff
@@ -206,11 +206,14 @@ void SlamToolbox::setROSInterfaces()
   scan_filter_sub_ =
     std::make_unique<message_filters::Subscriber<sensor_msgs::msg::LaserScan>>(
     shared_from_this().get(), scan_topic_, rmw_qos_profile_sensor_data);
+  scan_filter_sub_->unsubscribe();
```

Comment:
I don't think this PR works, did you test it? That patch worked in Nav2 because we later subscribe to it in the activated state https://github.com/SteveMacenski/navigation2/blob/f8b94c946c2bf841356b2b804e598a4fc10e4f8f/nav2_costmap_2d/plugins/obstacle_layer.cpp#L612, I don't think in this situation any laser data would ever get processed in the node.

Reply:
I did test it, but I tricked myself into thinking that it was working. I tried localization mode and for some reason it is still able to localize despite not receiving any laser data. It won't update the map though, and mapping mode won't work at all. Screen recording of my test.

https://user-images.githubusercontent.com/22382900/117504235-fe088c80-af36-11eb-8911-58869f6b3d90.mp4

The problem of "message filter dropping" warning being spammed still exists in foxy though:

https://user-images.githubusercontent.com/22382900/117504625-9bfc5700-af37-11eb-8fa3-84106f473c2f.mp4

Reply:
Well this isn't working at all, so I don't what to say - you've unsubscribed and didn't resubscribe, I would think that everything would be dropped.

Reply:
You're right, with my changes all the messages are dropped and I've made this PR unmergable. But the second video where it's working is the current foxy-devel branch and there is a continuous stream of "message filter dropping..." messages. This shows that when the laser scans are processed it outputs these messages even when slam_toolbox is working. I'm wondering if anything can be done about it. Other nodes which are using the laser scans (e.g rviz2, local/global costmap) don't have this problem.

Reply:
Thank you so much for your detailed feedback. Since this problem was occurring only in slam_toolbox, and not in other nodes using laser scan and TF, I had assumed that the fix had to be in slam_toolbox. But as you pointed out, `ros2 launch nav2_bringup tb3_simulation_launch.py slam:=True` does not have this problem. I will try to find what exactly in my setup is causing this problem based on your suggestions. Thanks again.